### ISSUE TITLE
UIDeviceFamily 6 breaks Mac Catalyst UI 

### ISSUE THREAD

**Issue by dhindrik** [dotnet/maui#11377]
### Description

Button BackgroundColor is not working at all. 

Measurement seems to be other also. But that is maybe not a bug.

### Steps to Reproduce

1. Change UIDeviceFamily to 6 in Info.plist for MacCatalyst
2. Add a button to UI.
3. Change the button color.

### Link to public reproduction project repository

https://github.com/dhindrik/AzServiceBus
(To run the repo you need to create a class names SecretKeys and add the missing constant with a dummy value)

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

macOS

### Affected platform versions

14.0

### Did you find any workaround?

It is possible to change UIDeviceFamily to for example 1, but then I am not allowed to upload it to App Store.

**Comment by dhindrik** [comment]
@jsuarezruiz if it was unclear, without setting UIDevice Family to 6 it is not possible to upload the app to App Store so I will say this is a high priority bug.

**Comment by drasticactions** [comment]
The default UIDeviceFamily for the MacCatalyst app is 2 (That's whats in the dotnet maccatalyst template and MAUIs); why are you setting to 1?

**Comment by drasticactions** [comment]
UIDeviceFamily 6 (The Mac Idiom) is not the default for MAUI. None of the controls were built with it in mind and I don't believe it's officially supported.

Having said that, the default buttons in the Mac idiom are bordered, and don't support background colors at all. You can try that in a regular Mac AppKit app with an NSButton. It's following that design pattern. If you want to color the background, you can turn the buttons into borderless ones, and then it should work.

<img width="1091" alt="スクリーンショット 2022-11-16 20 39 13" src="https://user-images.githubusercontent.com/898335/202189788-81cc4cc4-dfe5-405d-9c8e-56eeddc1e459.png">

```
        Microsoft.Maui.Handlers.ButtonHandler.Mapper.AppendToMapping("ButtonColor", (handler, view) =>
        {
#if MACCATALYST
            handler.PlatformView.Configuration = UIKit.UIButtonConfiguration.BorderlessButtonConfiguration;
            handler.PlatformView.BackgroundColor = UIKit.UIColor.Green;
#endif
        });
```

IMO though, if you want to match the Mac idiom, I would not color the buttons. If you want to, I would revert back to the Default, which is "2"

**Comment by dhindrik** [comment]
The default was both 1 and 2 and then I got an error about invalid values. And then I tried with 6 (optimize for Mac) and it worked to upload. But then UI was broken. Then I tried to change the values again and could not upload it when I set it to 1. I never realized that it should be only 2. Because 1 is "(Default) The app runs on iPhone and iPod touch devices." and 2 is "The app runs on iPad devices." according to the apple docs and they mention nothing about Mac at all. 

But now I changed to 2 and I was able to upload it to App Store :)
